code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/auth/config/GithubConfigManager.java

```diff
@@ -0,0 +1,208 @@
+
+    private static final DynamicBooleanProperty SECURITY = ArchaiusUtil.getBoolean("api.security.enabled");
+    private static final DynamicStringProperty GITHUB_CLIENT_ID = ArchaiusUtil.getString("api.auth.github.client.id");
+    private static final DynamicStringProperty GITHUB_ALLOWED_USERS = ArchaiusUtil.getString("api.auth.github.allow.users");
+    private static final DynamicStringProperty GITHUB_ALLOWED_ORGS = ArchaiusUtil.getString("api.auth.github.allow.orgs");
+
+    private JsonMapper jsonMapper;
+    private ObjectManager objectManager;
+    private GithubClient client;
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[] { GithubConfig.class };
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    protected Object createInternal(String type, ApiRequest request) {
+        if (!StringUtils.equals(GITHUB_CONFIG, request.getType())) {
+            return null;
+        }
+        Map<String, Object> config = jsonMapper.convertValue(request.getRequestBodyObject(), Map.class);
+        if (null != config.get("enabled")) {
+            createOrUpdateSetting("api.security.enabled", config.get("enabled"));
+        }
+        if (null != config.get("clientId")) {
+            createOrUpdateSetting("api.auth.github.client.id", config.get("clientId"));
+        }
+        if (null != config.get("clientSecret")) {
+            createOrUpdateSetting("api.auth.github.client.secret", config.get("clientSecret"));
+        }
+        if (null != config.get("allowUsers")) {
+            List<String> userList;
+            try {
+                userList = appendUserIds((List<String>) config.get("allowUsers"));
+            } catch (ClassCastException e) {
+                throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "allowedUsers should be a list");
+            }
+            createOrUpdateSetting("api.auth.github.allow.users", toCommaSeparatedString(userList));
+        }
+        if (null != config.get("allowOrganizations")) {
+            List<String> orgList;
+            try {
+                orgList = appendOrgIds((List<String>) config.get("allowOrganizations"));
+            } catch (ClassCastException e) {
+                throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "allowedOrganizations should be a list");
+            }
+            createOrUpdateSetting("api.auth.github.allow.orgs", toCommaSeparatedString(orgList));
+        }
+
+        return new Object();
+    }
+
+    protected List<String> appendUserIds(List<String> usernames) {
+        if(usernames == null) {
+            return null;
+        }
+        return Lists.transform(usernames, new Function<String, String> () {
+
+            @Override
+            public String apply(String username) {
+                GithubAccountInfo userInfo = client.getUserIdByName(username);
+                if(userInfo == null) {
+                    throw new RuntimeException("Invalid username: " + username);
```

Comment:
Runtime is a 500 error which is a server side error.  This should be a ClientVisibleException.  Probably with a nice message saying invalid github username.